include/ledmatrixgfx.h

```diff
@@ -126,10 +126,11 @@ class LEDMatrixGFX : public GFXBase
 
     virtual uint16_t xy(uint16_t x, uint16_t y) const override
     {
-        if (x >= 0 && x < _width && y >= 0 && y < _height)
+        // Note the x,y are unsigned so can't be less than zero
+        if (x < _width && y < _height)
```

Comment:
When porting code from other systems, this means each effect ends up implementing its own xy and drawpixels that mask because the systems ones (unique to us and unlike smartmatrix or marc merlin's or soulmate or ....) have to be avoided.

I know this isn't changing in this submit. Just pointing out that it's a pain point to effects developers.

Reply:
I've read that several times and each time I'm less sure what you mean!  Effects don't need to implement xy, they consume it.  It's handled by the virtual implementation on the GFXBase class instance.

Can you explain more?  Or differently?  Or type slower so I can understand? :-)

Reply:
Sorry. I can be scatter-brained. You're correct, it's not xy() itself that gets reimplemented; it's everything using it, which is a lot. (I also see that I'm logged into the "wrong" account for this project, but I suspect you recognize this is Robert)

If you're porting an effect from another environment (let's say Animatrix, though it's just an example), that code can call drawCircle, drawLine or whatever with X and Y that are "off screen" with immunity. No wrapping is done and no error is enforced; it's simply masked. (clamp or integer modulo would be fast ways to implement it) It's not an error case. I thus wound up implementing (copy-pasting) a lot of drawCircle-style implementations because the system one would fall into this throw. There were a few that called drawLine near the edges that would similarly fall into this throw.

My request here is to make this a debugX warning instead of a hard error. (Clamping it so you have a superpixel in one corner isn't great, but it's at least a hint to the effect developer.) In xy() it's only annoying, but in the drawing primitives, it's a big deal.